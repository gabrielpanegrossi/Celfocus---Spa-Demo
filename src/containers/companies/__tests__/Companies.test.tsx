import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Companies from '../index';

jest.mock('react-query', () => ({
  useQuery: () => ({
    isLoading: false,
    error: undefined,
    data: [{ id: 1, name: 'Local Public Office', vatin: '500754012' }],
  }),
}));

describe('ContainerHeader', () => {
  beforeEach(() => {
    render(
      <Router>
        <Companies />
      </Router>
    );
  });

  it('should match snpashot', () => {
    const container = render(
      <Router>
        <Companies />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render', () => {
    expect(screen.queryByRole('table')).toBeInTheDocument();
  });

  it('should render fetched info on table', async () => {
    expect(screen.queryByRole('table')).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.queryByText('Local Public Office')).toBeInTheDocument();
      expect(screen.queryByText('500754012')).toBeInTheDocument();
    });
  });

  it('should navigate to company page', async () => {
    const user = userEvent.setup();

    await waitFor(() => {
      expect(screen.queryByText('Local Public Office')).toBeInTheDocument();
    });

    user.click(screen.queryByText('Local Public Office'));

    await waitFor(() => {
      expect(global.window.location.href).toContain('/company/1');
    });
  });
});
