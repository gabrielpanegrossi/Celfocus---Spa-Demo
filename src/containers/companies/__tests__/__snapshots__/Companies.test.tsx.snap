// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerHeader should match snpashot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKXOVf bGBQSC"
      >
        <header
          class="sc-bczRLJ bXXmoA"
        >
          <div
            class="sc-gsnTZi jGtclP"
          />
          <div
            class="sc-gsnTZi jGtclP"
          >
            <h1
              class="sc-hKMtZM cfLWee"
            >
              Companies
            </h1>
          </div>
          <div
            class="sc-gsnTZi jGtclP"
          />
        </header>
        <div
          class="sc-iBkjds fzLwdz"
        >
          <table
            class="sc-eCYdqJ vSbeb"
          >
            <thead>
              <tr>
                <th>
                  Company name
                </th>
                <th>
                  vatin
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <a
                    href="/company/1"
                  >
                    Local Public Office
                  </a>
                </td>
                <td>
                  500754012
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-gKXOVf bGBQSC"
      >
        <header
          class="sc-bczRLJ bXXmoA"
        >
          <div
            class="sc-gsnTZi jGtclP"
          />
          <div
            class="sc-gsnTZi jGtclP"
          >
            <h1
              class="sc-hKMtZM cfLWee"
            >
              Companies
            </h1>
          </div>
          <div
            class="sc-gsnTZi jGtclP"
          />
        </header>
        <div
          class="sc-iBkjds fzLwdz"
        >
          <table
            class="sc-eCYdqJ vSbeb"
          >
            <thead>
              <tr>
                <th>
                  Company name
                </th>
                <th>
                  vatin
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <a
                    href="/company/1"
                  >
                    Local Public Office
                  </a>
                </td>
                <td>
                  500754012
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKXOVf bGBQSC"
    >
      <header
        class="sc-bczRLJ bXXmoA"
      >
        <div
          class="sc-gsnTZi jGtclP"
        />
        <div
          class="sc-gsnTZi jGtclP"
        >
          <h1
            class="sc-hKMtZM cfLWee"
          >
            Companies
          </h1>
        </div>
        <div
          class="sc-gsnTZi jGtclP"
        />
      </header>
      <div
        class="sc-iBkjds fzLwdz"
      >
        <table
          class="sc-eCYdqJ vSbeb"
        >
          <thead>
            <tr>
              <th>
                Company name
              </th>
              <th>
                vatin
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  href="/company/1"
                >
                  Local Public Office
                </a>
              </td>
              <td>
                500754012
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
